src/Bicep.Core/Emit/ParametersFileTemplateWriter.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Semantics;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+
+namespace Bicep.Core.Emit
+{
+    public class ParametersFileTemplateWriter : TemplateWriter
+    {
+        public ParametersFileTemplateWriter(SemanticModel semanticModel, EmitterSettings settings): base(semanticModel, settings)
+        {
+            this.context = new EmitterContext(semanticModel, settings);
+            this.settings = settings;
+        }
+
+        private readonly EmitterContext context;
+        private readonly EmitterSettings settings;
+
+        public new void Write(JsonTextWriter writer)
+        {
+            // Template is used for calcualting template hash, template jtoken is used for writing to file.
+            var templateJToken = GenerateTemplate();
```

Comment:
I wonder if we need to generate a template here. Since we have access to the semantic model, we should be able to generate parameters using the parameter type information.

Reply:
Yeah, I agree.